examples/cli_workflow/README.md

```diff
@@ -0,0 +1,288 @@
+>       init_args:
+>         target_count: 10_000
+>     - cellarium.ml.transforms.Log1p
+```
+
+Change `OnePassMeanVarStd`'s algorithm to `shifted_data`:
+
+```diff
+<       algorithm: naive
+---
+>       algorithm: shifted_data
+```
+
+### data
+
+Configure the `DistributedAnnDataCollection`. Here we validate `obs` columns that are used by the transforms and the model (`total_mrna_umis`):
+
+```diff
+<       filenames: null
+<       shard_size: null
+<       max_cache_size: 1
+<       obs_columns_to_validate: null
+---
+>       filenames: https://storage.googleapis.com/dsp-cellarium-cas-public/test-data/test_{0..3}.h5ad
+>       shard_size: 100
+>       max_cache_size: 2
+>       obs_columns_to_validate:
+>         - total_mrna_umis
+```
+
+Configure the `DataLoader`. `batch_keys` have to include all input arguments to the transforms and the model. For example, `NormalizeTotal`'s arguments are `x_ng` and `total_mrna_umis_n`, `Log1p`'s argument is `x_ng`, and `OnePassMeanVarStd`'s arguments are `x_ng` and `var_names_g`:
+
+```diff
+<   batch_keys: null
+<   batch_size: 1
+<   num_workers: 0
+---
+>   batch_keys:
+>     x_ng:
+>       attr: X
+>       convert_fn: cellarium.ml.utilities.data.densify
+>     var_names_g:
+>       attr: var_names
+>     total_mrna_umis_n:
+>       attr: obs
+>       key: total_mrna_umis
+>   batch_size: 100
+>   num_workers: 2
+```
+
+## `IncrementalPCA`
+
+Generate a default config file:
+
+```
+cellarium-ml incremental_pca fit --print_config > ipca_train_config.yaml
+```
+
+Below we highlight the changes made to the default configuration file.
+
+### train
+
+Change the number of devices and set the path for logs and weights:
+
+```diff
+<   devices: auto
+<   default_root_dir: null
+---
+>   devices: 2
+>   default_root_dir: /tmp/test_examples/ipca
+```
+
+### model
+
+Add `NormalizeTotal` and `Log1p`, and `ZScore` transforms. Note, that `mean_g`, `std_g`, and `var_names_g` of `ZScore` transform are loaded from the `OnePassMeanVarStd` checkpoint:
+
+```diff
+<   transforms: null
+---
+>   transforms:
+>     - class_path: cellarium.ml.transforms.NormalizeTotal
+>       init_args:
+>         target_count: 10_000
+>     - cellarium.ml.transforms.Log1p
+>     - class_path: cellarium.ml.transforms.ZScore
+>       init_args:
+>         mean_g:
+>           !CheckpointLoader
+>           file_path: /tmp/test_examples/onepass/lightning_logs/version_0/checkpoints/epoch=0-step=2.ckpt
+>           attr: model.mean_g
+>           convert_fn: null
+>         std_g:
+>           !CheckpointLoader
+>           file_path: /tmp/test_examples/onepass/lightning_logs/version_0/checkpoints/epoch=0-step=2.ckpt
+>           attr: model.std_g
+>           convert_fn: null
+>         var_names_g:
+>           !CheckpointLoader
+>           file_path: /tmp/test_examples/onepass/lightning_logs/version_0/checkpoints/epoch=0-step=2.ckpt
+>           attr: model.var_names_g
+>           convert_fn: numpy.ndarray.tolist
+```
+
```

Comment:
Maybe add a note below that cellarium-ml does not perform any validation on the content of data loaded from the checkpoints, or whether it is consistent with the rest of the configuration. For example, the mean and std were calculate in a prior stop from data that was subject to normalize total and log1p transform. If the user inadvertently forgets to perform the same transforms here before z-scoring, the workflow will run w/o any error though will produce wrong results.